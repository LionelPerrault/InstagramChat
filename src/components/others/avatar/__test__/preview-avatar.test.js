import React from 'react'
import { create } from 'react-test-renderer'
import PreviewAvatar from '../preview-avatar'
import { shallow } from 'enzyme'

describe('Preview-Avatar Component', () => {
  const props = {
    previewAvatar: '/images/location.jpg',
    back() {},
    upload() {},
  }

  it('should match snapshot', () => {
    const tree = create(<PreviewAvatar {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should perform some click events', () => {
    const wrapper = shallow(<PreviewAvatar {...props} />)

    wrapper.find('SecondaryButton').simulate('click')
    wrapper.find('PrimaryButton').simulate('click')
  })
})
