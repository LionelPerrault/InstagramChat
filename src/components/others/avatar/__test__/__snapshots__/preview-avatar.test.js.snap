// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview-Avatar Component should match snapshot 1`] = `
Array [
  <div
    className="overlay"
    onClick={null}
    style={
      Object {
        "cursor": "inherit",
        "opacity": 0.5,
      }
    }
  />,
  <div
    className="preview_avatar modal"
  >
    <div
      style={
        Object {
          "animation": "bcCCNc 300ms ease 0s 1 normal none running",
          "backfaceVisibility": "visible",
        }
      }
    >
      <div
        className="modal_header"
      >
        <span
          className="title"
        >
          Change Avatar
        </span>
      </div>
      <div
        className="c_a_middle"
      >
        <img
          src="/images/location.jpg"
        />
      </div>
      <div
        className="c_a_bottom modal_bottom"
      >
        <a
          className="sec_btn"
          href="#"
          onClick={[Function]}
        >
          Cancel
        </a>
        <a
          className="pri_btn c_a_add"
          href="#"
          onClick={[Function]}
        >
          Change avatar
        </a>
      </div>
    </div>
  </div>,
]
`;
